orchestrator-salt/src/main/resources/salt/salt/cloudera/manager/start.sls

```diff
@@ -8,6 +9,27 @@ init-cloudera-manager-db:
         - user: {{ cloudera_manager.cloudera_manager_database.connectionUserName }}
         - pass: {{ cloudera_manager.cloudera_manager_database.connectionPassword }}
 
+# We need to restart the CM if the fromtend URL has been changed
+{% if salt['pillar.get']('gateway:userfacingfqdn') is defined and salt['pillar.get']('gateway:userfacingfqdn')|length > 1 %}
+update_frontend_url_of_cm:
+  file.replace:
+      - name: /etc/cloudera-scm-server/cm.settings
+      - pattern: 'setsettings FRONTEND_URL.*'
+      - repl: 'setsettings FRONTEND_URL https://{{ salt['pillar.get']('gateway:userfacingfqdn') }}/'
+      - unless: grep "^setsettings FRONTEND_URL https://{{ salt['pillar.get']('gateway:userfacingfqdn') }}/$" /etc/cloudera-scm-server/cm.settings
+
+restart_cm_after_fronted_url_change:
+  service.running:
+    - name: cloudera-scm-server
+    - enable: True
```

Comment:
I think we can omit this one as `start_server` already contains it, no reason to duplicate the same setting

Reply:
ok